### PR TITLE
Better router mapping

### PR DESCRIPTION
**Pull request by tzaffi**
## PLEASE DISREGARD THIS PR

### CODE REVIEW

File: pyteal/stack_frame.py [hunk 1]
```diff
@@ -365,9 +407,16 @@ def _walk_asts(
                     walker_args = [expr.value]
                     if expr.index_expression:
                         walker_args.append(expr.index_expression)
+                case SubroutineCall():
+                    subdef = expr.subroutine.get_declaration_by_option(False)
+                    if subdef:
+                        walker_args.append(subdef)
+                    subdef = expr.subroutine.get_declaration_by_option(True)
+                    if subdef:
+                        walker_args.append(subdef)
```
Comment (tzaffi) [inline pyteal/stack_frame.py:416]: @ahangsu - introducing these lines of code into the sourcemapper, actually changed the resulting teal. I'm trying to understand why.

File: pyteal/compiler/flatten.py [hunk 2]
```diff
@@ -123,7 +144,13 @@ def flattenSubroutines(
             if isinstance(stmt, TealLabel):
                 stmt.getLabelRef().addPrefix(labelPrefix)
 
-        combinedOps.append(TealLabel(None, LabelReference(label), comment))
+        dexpr: Expr | None = None
+        try:
+            dexpr = subroutine.declarations.get_declaration()
```
Comment (tzaffi) [inline pyteal/compiler/flatten.py:149]: deprecated

File: pyteal/stack_frame.py (hunk 1 above)
Comment (ahangsu) [reply pyteal/stack_frame.py:416]: seems fixed? For we introduced a fix #684 and the problem should go away

File: pyteal/compiler/flatten.py (hunk 2 above)
Comment (ahangsu) [reply pyteal/compiler/flatten.py:149]: I believe this is also relieved in #650

File: pyteal/stack_frame.py (hunk 1 above)
Comment (tzaffi) [reply pyteal/stack_frame.py:416]: Absolutely. Please disregard this stale comment. Also, I'm about to close this draft in favor of #689

**Pull request closed without merging**